### DISCUSSION TITLE
Long Response Time for Livewire AJAX Calls

### DISCUSSION THREAD

**Discussion by okonmelon** [livewire/livewire#5750] [1. Help] [answered]
I'm trying to use Laravel with Livewire to do CRUD operations without having to refresh the page each time an operation occurs. My problem is that all the AJAX requests that Livewire is doing are taking around 1.5 seconds. I had found a discussion with a similar issue, but it wasn't that helpful since the cause was just that their database was hosted far away so each database query was taking a while. That discussion at least introduced me to the Debugbar and let me rule out database queries as being the source of the delay, query times shown below.
![image](https://github.com/livewire/livewire/assets/77363864/0657b625-10a1-4139-a3a2-157948a209a0)

Anyway, I decided to add a Debugbar measure which starts in the component's boot function and stops in it's booted function just to see if what's causing the delay is happening before or after the component has been booted. The image below shows the results.
![image](https://github.com/livewire/livewire/assets/77363864/c810690f-b1cb-49e5-860f-33c97ad32dbf)
~40ms of the 880ms of the Application time are either during or after the livewire component's boot function. The My Function time is the time it took my delete method to run.

Any ideas what's causing Livewire to take so long just to get to it's boot function?

**Comment by onemoreahmad** [comment]
Please show us your delete method, we can't help without some code provided. 
Or try to comment everything in your delete method and return just simple string and check the speed again.

**Reply by okonmelon** [reply]
Here's the result with the entirety of the delete method's body commented out.

![image](https://github.com/livewire/livewire/assets/138519754/d35f6b00-e98b-42cb-af7a-1a816f10a7c5)

As stated before, it is not only taking this long on the delete method, that whole second prior to the livewire boot function in the "Application" time shows up on page load and on all AJAX calls, delete was just the method I used in order for livewire to execute an AJAX call to show the times and to show that the actual method's execution time was pretty miniscule compared to the response time (~24.9ms method execution time vs 1s response time).

**Reply by onemoreahmad** [reply, marked as answer]
To better debug, I would install a fresh Laravel and install only Livewire and Debugbar and nothing else and test it again. 
I had a similar issue few months ago and when I install everything from scratch I discovered that two ` blade icons packages` taking too much to load, so check packages and what service providers you're booting causing this load time.

**Reply by okonmelon** [reply]
Ah, I had recently installed a blade icon package. Just removed it and speeds are already a lot better, thank you!

**Reply by onemoreahmad** [reply]
No problem!
